packages/vscode-extension/src/debug/depsChecker/funcToolChecker.ts

```diff
@@ -51,187 +46,210 @@ export class FuncToolChecker implements IDepsChecker {
-        // this._telemetry.sendEvent(DepsCheckerEvent.funcV1Installed);
-        // this._telemetry.sendUserErrorEvent(
-        //   DepsCheckerEvent.checkFunc,
-        //   TelemtryMessages.funcV1Installed
-        // );
-        throw new DepsCheckerError(
-          Messages.needReplaceWithFuncCoreToolV3.replace("@NameVersion", installedNameWithVersion),
-          defaultHelpLink
-        );
-      case FuncVersion.v2:
-        // TODO: should send this event per user.
-        // this._telemetry.sendEvent(DepsCheckerEvent.funcV2Installed);
-        return installed;
-      case FuncVersion.v3:
-        // TODO: should send this event per user.
-        // this._telemetry.sendEvent(DepsCheckerEvent.funcV3Installed);
-        return installed;
-      default:
-        return !installed;
+    const isGlobalFuncInstalled: boolean = await this.isGlobalFuncInstalled();
+    const isPortableFuncInstalled: boolean = await this.isPortableFuncInstalled();
+
+    if (isGlobalFuncInstalled) {
+      this._telemetry.sendEvent(DepsCheckerEvent.funcAlreadyInstalled, {
+        "global-func-version": `${await this.queryFuncVersionSilently("func")}`,
+      });
     }
+    if (isPortableFuncInstalled) {
+      // avoid missing this event after first installation 60 days
+      this._telemetry.sendEvent(DepsCheckerEvent.funcInstallCompleted);
+    }
+
+    return isPortableFuncInstalled || isGlobalFuncInstalled;
+  }
+
+  public async isPortableFuncInstalled(): Promise<boolean> {
+    const portableFuncVersion = await this.queryFuncVersionSilently(
+      FuncToolChecker.getPortableFuncExecPath()
+    );
+    return portableFuncVersion !== null && supportedVersions.includes(portableFuncVersion);
+  }
+
+  public async isGlobalFuncInstalled(): Promise<boolean> {
+    const globalFuncVersion = await this.queryFuncVersionSilently("func");
+    return globalFuncVersion !== null && supportedVersions.includes(globalFuncVersion);
   }
 
   public async install(): Promise<void> {
     if (!(await this.hasNPM())) {
-      // provided with Learn More link if npm doesn't exist.
-      this._telemetry.sendUserErrorEvent(
-        DepsCheckerEvent.funcInstall,
-        TelemtryMessages.NPMNotFound
-      );
-      throw new DepsCheckerError(
-        Messages.needInstallFuncCoreTool.replace("@NameVersion", installedNameWithVersion),
-        defaultHelpLink
-      );
+      this.handleNpmNotFound();
+    }
+
+    await this.cleanup();
+    await this.installFunc();
+
+    if (!(await this.validate())) {
+      await this.handleInstallFuncFailed();
     }
 
+    this._telemetry.sendEvent(DepsCheckerEvent.funcInstallCompleted);
     await this._logger.info(
-      Messages.startInstallFunctionCoreTool.replace("@NameVersion", installedNameWithVersion)
+      Messages.finishInstallFunctionCoreTool.replace("@NameVersion", displayFuncName)
+    );
+  }
+
+  private async handleInstallFuncFailed(): Promise<void> {
+    await this.cleanup();
```

Comment:
Need to handle the exception during cleanup, otherwise below system error event won't be sent?

Reply:
Fixed